src/Type/PostObject/Mutation/PostObjectUpdate.php

```diff
@@ -0,0 +1,172 @@
+
+use GraphQLRelay\Relay;
+use WPGraphQL\Type\PostObject\PostObjectMutation;
+use WPGraphQL\Types;
+
+/**
+ * Class PostObjectUpdate
+ *
+ * @package WPGraphQL\Type\PostObject\Mutation
+ */
+class PostObjectUpdate {
+
+	/**
+	 * Holds the mutation field definition
+	 *
+	 * @var array $mutation
+	 */
+	private static $mutation;
+
+	/**
+	 * Defines the Update mutation for PostTypeObjects
+	 *
+	 * @param \WP_Post_Type $post_type_object
+	 *
+	 * @return array|mixed
+	 */
+	public static function mutate( \WP_Post_Type $post_type_object ) {
+
+		if ( null === self::$mutation ) {
+			self::$mutation = [];
+		}
+
+		if ( ! empty( $post_type_object->graphql_single_name ) && empty( self::$mutation[ $post_type_object->graphql_single_name ] ) ) :
+
+			/**
+			 * Set the name of the mutation being performed
+			 */
+			$mutation_name = 'update' . ucwords( $post_type_object->graphql_single_name );
+
+			self::$mutation[ $post_type_object->graphql_single_name ] = Relay::mutationWithClientMutationId( array(
+				'name'        => esc_html( $mutation_name ),
+				// translators: The placeholder is the name of the post type being updated
+				'description' => sprintf( esc_html__( 'Updates %1$s objects', 'wp-graphql' ), $post_type_object->graphql_single_name ),
+				'inputFields' => self::input_fields( $post_type_object ),
+				'outputFields' => array(
+					$post_type_object->graphql_single_name => array(
+						'type'    => Types::post_object( $post_type_object->name ),
+						'resolve' => function( $payload ) {
+							return get_post( $payload['postObjectId'] );
+						},
+					),
+				),
+				'mutateAndGetPayload' => function( $input ) use ( $post_type_object, $mutation_name ) {
+
+					$id_parts = ! empty( $input['id'] ) ? Relay::fromGlobalId( $input['id'] ) : null;
+					$existing_post = get_post( absint( $id_parts['id'] ) );
+
+					/**
+					 * If there's no existing post, throw an exception
+					 */
+					if ( empty( $id_parts['id'] ) || false === $existing_post ) {
+						// translators: the placeholder is the name of the type of post being updated
+						throw new \Exception( sprintf( __( 'No %1$s could be found to update', 'wp-graphql' ), $post_type_object->graphql_single_name ) );
+					}
+
+					if ( $post_type_object->name !== $existing_post->post_type ) {
+						// translators: The first placeholder is an ID and the second placeholder is the name of the post type being edited
+						throw new \Exception( sprintf( __( 'The id %1$d is not of the type "%2$s"', 'wp-graphql' ), $id_parts['id'], $post_type_object->name ) );
+					}
+
+					/**
+					 * Stop now if a user isn't allowed to edit posts
+					 */
+					if ( ! current_user_can( $post_type_object->cap->edit_posts ) ) {
+						// translators: the $post_type_object->graphql_single_name placeholder is the name of the object being mutated
+						throw new \Exception( sprintf( __( 'Sorry, you are not allowed to update a %1$s', 'wp-graphql' ), $post_type_object->graphql_single_name ) );
+					}
```

Comment:
I wonder if it would make sense to abstract these cap checks into a `postObjectMutateCheck` utility class or something. Just a thought. Not required before merge.